artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java

```diff
@@ -2245,6 +2379,155 @@ private synchronized JournalLoadInformation load(final LoaderCallback loadManage
       }
    }
 
+
+   @Override
+   public void processBackupCleanup() {
+      if (journalRetentionFolder != null && (journalRetentionMaxFiles > 0 || journalRetentionPeriod > 0)) {
+
+         FilenameFilter fnf = new FilenameFilter() {
+            @Override
+            public boolean accept(final File file, final String name) {
+               return name != null && name.endsWith("." + filesRepository.getFileExtension());
+            }
+         };
+
+
+         if (journalRetentionPeriod > 0) {
+            String[] fileNames = journalRetentionFolder.list(fnf);
+            Arrays.sort(fileNames);
+
+            GregorianCalendar calendar = this.calendarThreadLocal.get();
+            calendar.setTimeInMillis(System.currentTimeMillis() - journalRetentionUnit.toMillis(journalRetentionPeriod));
```

Comment:
Why doing the conversion to ms every time, why not convert to ms just once all the way up at point of reading config

Reply:
ok.. fair enough.. I can do that